msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OnBehalfOfRequest.java

```diff
@@ -30,20 +29,21 @@ class OnBehalfOfRequest extends MsalRequest {
     }
 
     private static OAuthAuthorizationGrant createAuthenticationGrant(OnBehalfOfParameters parameters) {
+        Map<String, List<String>> params = new LinkedHashMap<>();
 
-        AuthorizationGrant jWTBearerGrant;
-        try {
-            jWTBearerGrant = new JWTBearerGrant(SignedJWT.parse(parameters.userAssertion().getAssertion()));
-        } catch (Exception e) {
-            throw new MsalClientException(e);
-        }
-
-        Map<String, List<String>> params = new HashMap<>();
+        params.put(GrantConstants.GRANT_TYPE_PARAMETER, Collections.singletonList(GrantConstants.JWT_BEARER));
         params.put("requested_token_use", Collections.singletonList("on_behalf_of"));
+
         if (parameters.claims() != null) {
             params.put("claims", Collections.singletonList(parameters.claims().formatAsJSONString()));
         }
 
-        return new OAuthAuthorizationGrant(jWTBearerGrant, String.join(SCOPES_DELIMITER, parameters.scopes()), params);
+        try {
+            params.put(GrantConstants.ASSERTION_PARAMETER, Collections.singletonList(SignedJWT.parse(parameters.userAssertion().getAssertion()).getParsedString()));
```

Comment:
issue: We should never parse tokens. Can we get a bug to remove this?

Reply:
Removed in the latest commit.

When it was using the dependency's `JWTBearerGrant` class the constructor needed a `SignedJWT` object created from the assertion string, and I just copied that without thinking that the assertion string was all we needed.